### PR TITLE
Add support for JSON parameterization of build. And microfrontends.json -> frontend.json

### PR DESCRIPTION
**Pull request by brandones**
This is to support OpenMRS SDK integration. An implementer will be able to specify these properties in their distribution.

### CODE REVIEW

File: packages/tooling/openmrs/src/cli.ts [hunk 1]
```diff
@@ -178,6 +178,8 @@ yargs.command(
         "Determines if the output directory should be cleaned before the run."
       )
       .default("fresh", false)
+      .string("build-config")
+      .describe("build-config", "Path to a frontend build/assemble config JSON")
```
Comment (FlorianRappl) [inline packages/tooling/openmrs/src/cli.ts:182]: What's the default here? This should be `optional`.

File: packages/tooling/openmrs/src/cli.ts [hunk 2]
```diff
@@ -225,11 +228,8 @@ yargs.command(
       .default("registry", "https://registry.npmjs.org/")
       .describe("registry", "The NPM registry used for getting the packages.")
       .string("config")
-      .default("config", "microfrontends.json")
-      .describe(
-        "config",
-        "The configuration for gathering the list of microfrontends to include."
-      )
+      .default("config", "frontend.json")
```
Comment (FlorianRappl) [inline packages/tooling/openmrs/src/cli.ts:231]: I think this name is not really right. Beforehand the name was more fitting. If you want to fit the general use case why don't you default to `openmrs.json`?

File: packages/tooling/openmrs/src/commands/build.ts [hunk 3]
```diff
@@ -1,26 +1,48 @@
...
+  configUrls: Array<string>;
+  importmap: string;
+  spaPath: string;
+}
+
+function loadBuildConfig(buildConfigPath?: string): BuildConfig {
+  if (buildConfigPath) {
+    return JSON.parse(readFileSync(buildConfigPath, "utf8"));
+  } else {
+    return {} as BuildConfig;
```
Comment (FlorianRappl) [inline packages/tooling/openmrs/src/commands/build.ts:33]: If the file was invalid or not existing we should error out.

File: packages/tooling/openmrs/src/cli.ts [hunk 4]
```diff
@@ -163,7 +163,7 @@ yargs.command(
 
 yargs.command(
   "build",
-  "Builds a new app shell using the provided configuration.",
+  "Builds a new app shell.",
```
Comment (FlorianRappl) [inline packages/tooling/openmrs/src/cli.ts:166]: Right now this does a bit more than that. Also, it is now inconsistent with `debug`, so some of these changes (using a config file) should be brought in there, too.

File: packages/tooling/openmrs/src/cli.ts (hunk 1 above)
Comment (brandones) [reply packages/tooling/openmrs/src/cli.ts:182]: The default is undefined; yargs treats all arguments as optional unless [`demandOption`](https://github.com/yargs/yargs/blob/master/docs/api.md#demandOption) is used.

File: packages/tooling/openmrs/src/cli.ts (hunk 2 above)
Comment (brandones) [reply packages/tooling/openmrs/src/cli.ts:231]: The application we're working on is being marketed as "OpenMRS Frontend 3.0," no longer as "OpenMRS Microfrontends." And this change makes this JSON file no longer only a list of microfrontends, but rather describe the configuration for the frontend application generally. `openmrs.json` would introduce namespace clash with server-side configuration files.

File: packages/tooling/openmrs/src/commands/build.ts (hunk 3 above)
Comment (brandones) [reply packages/tooling/openmrs/src/commands/build.ts:33]: Yes, that is how it behaves. `readFileSync` throws an error if the file doesn't exist, and `JSON.parse` throws an error if the file is invalid. `{}` is returned only in the case that no build config is provided.

File: packages/tooling/openmrs/src/cli.ts (hunk 4 above)
Comment (brandones) [reply packages/tooling/openmrs/src/cli.ts:166]: Yeah, definitely. For now, any suggestion about how to describe what `build` does? It builds the app shell, and what else?

File: packages/tooling/openmrs/src/cli.ts (hunk 2 above)
Comment (FlorianRappl) [reply packages/tooling/openmrs/src/cli.ts:231]: Not sure I agree.

1. This is only for the default name; the user can give the file any name
2. Why would this tooling care about any other tooling (that is not directly used by it) or how something is marketed?

I'd say let's make the name `spa-build-config.json`.

Comment (brandones) [reply packages/tooling/openmrs/src/cli.ts:231]: Sure, Florian.

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks! :beers:

**Pull request merged**